packages/core/src/common/url.ts

```diff
@@ -14,9 +14,47 @@
  * limitations under the License.
  */
 
-const expression = /[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/gi;
-const regex = new RegExp(expression);
-
+const re_weburl = new RegExp(
```

Comment:
Should have test cases to cover this regular expression. And regular expression are **values with state**. It is recommended to create different instances instead of reuse the same one (or reset its state before another calling) to avoid potential problems.